core/packages/contracts/src/Types.sol

```diff
@@ -2,18 +2,56 @@
 // SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
 pragma solidity 0.8.20;
 
-type ParaID is uint32;
+type ParaID is uint256;
 
-using {eq as ==, ne as !=, isNone} for ParaID global;
+using {ParaIDEq as ==, ParaIDNe as !=} for ParaID global;
 
-function eq(ParaID a, ParaID b) pure returns (bool) {
+function ParaIDEq(ParaID a, ParaID b) pure returns (bool) {
     return ParaID.unwrap(a) == ParaID.unwrap(b);
 }
 
-function ne(ParaID a, ParaID b) pure returns (bool) {
-    return !eq(a, b);
+function ParaIDNe(ParaID a, ParaID b) pure returns (bool) {
+    return !ParaIDEq(a, b);
 }
 
-function isNone(ParaID a) pure returns (bool) {
-    return ParaID.unwrap(a) == 0;
+struct Channel {
+    OperatingMode mode;
+    uint64 inboundNonce;
+    uint64 outboundNonce;
+    address agent;
+    uint256 fee;
+    uint256 reward;
+}
+
+// Inbound message from a Polkadot parachain (via BridgeHub)
+struct InboundMessage {
+    ParaID origin;
+    uint64 nonce;
+    Command command;
+    bytes params;
+}
+
+enum OperatingMode {
+    Normal,
+    RejectingOutboundMessages
+}
```

Comment:
Is mode `RejectingInboundMessages` nessessary here?

Reply:
My thinking is that if on BridgeHub we update the pallet's operating mode to `RejectingOutboundMessages`, it will have the same effect of preventing any more messages being sent to Ethereum.

Reply:
My concern is assuming some security bug exist in our bridge contract which allows malicious relayer/attacker to construct transaction manually(i.e. not from BridgeHub pallets) and steal money from the protocol.

Reply:
Assuming there a bug in code, there is no way to guard against this kind of attack, since if a malicious attacker can manually construct a transaction and have it pass verification by the Gateway, then they can take over entire control of the Gateway using the `upgrade` handler.